cilium/cmd/map_get_history.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	daemonAPI "github.com/cilium/cilium/api/v1/client/daemon"
+	"github.com/cilium/cilium/api/v1/models"
+	"github.com/cilium/cilium/pkg/api"
+	"github.com/cilium/cilium/pkg/command"
+
+	"github.com/spf13/cobra"
+)
+
+var mapGetHistory = &cobra.Command{
+	Use:     "history <name>",
+	Short:   "Display BPF map operations history",
+	Example: "cilium map get cilium_ipcache history",
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) == 0 {
+			Fatalf("map name must be specified")
+		}
+
+		params := daemonAPI.NewGetMapNameHistoryParams().WithName(
+			args[0]).WithTimeout(api.ClientTimeout)
+
+		resp, err := client.Daemon.GetMapNameHistory(params)
+		if err != nil {
+			Fatalf(err.Error())
+		}
+
+		history := resp.Payload
+		if history == nil {
+			return
+		}
+
+		if command.OutputJSON() {
+			if err := command.PrintOutput(history); err != nil {
+				os.Exit(1)
+			}
+		} else {
+			printMapHistory(history)
+		}
+	},
+}
+
+func printMapHistory(history *models.BPFMapHistory) {
+	w := tabwriter.NewWriter(os.Stdout, 5, 0, 3, ' ', 0)
```

Comment:
And here, you would do a sort by time since the proposal that I've said with the `i++` would make the history to be out of order.

Reply:
I think I will do sorting on the API side.